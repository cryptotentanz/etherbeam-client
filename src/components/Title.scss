@use '../styles/resources/sizes';

.my-title {
  display: inline-flex;
  align-items: center;
  font-weight: bold;
  font-style: italic;
  font-size: sizes.$size-xl;
  height: 100%;

  &-label {
    font-size: sizes.$size-s;
  }

  &-img {
    width: sizes.$size;
    height: sizes.$size;
  }
}
