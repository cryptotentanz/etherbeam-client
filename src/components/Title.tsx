import React, { FC } from 'react'
import './Title.scss'

type LabelBreakpointProp = 'xs' | 's' | 'm' | 'l'

interface TitleProps {
  labelBreakpoint?: LabelBreakpointProp
}

const Title: FC<TitleProps> = ({ labelBreakpoint }: TitleProps) => {
  const labelClassName = ['my-title-label', labelBreakpoint ? `my-d-min-${labelBreakpoint}` : ''].join(' ')

  return (
    <span className="my-title">
      <img className="my-title-img" src="/images/ethereum.png" />
      <span className={labelClassName}>Etherbeam</span>
    </span>
  )
}

export default Title
