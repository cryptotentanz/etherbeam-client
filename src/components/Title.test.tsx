import React from 'react'
import { shallow, ShallowWrapper } from 'enzyme'
import Title from './Title'

describe('<Title />', () => {
  let component: ShallowWrapper

  describe('Default', () => {
    beforeAll(() => {
      component = shallow(<Title />)
    })

    it('should match snapshot', () => expect(component.render()).toMatchSnapshot())
  })

  describe('Label breakpoint', () => {
    beforeAll(() => {
      component = shallow(<Title labelBreakpoint="m" />)
    })

    it('should render', () => expect(component.find('span').at(0).children().at(1).hasClass('my-d-min-m')).toBeTruthy())
  })
})
