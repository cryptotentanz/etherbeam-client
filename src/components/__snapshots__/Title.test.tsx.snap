// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Title /> Default should match snapshot 1`] = `
<span
  class="my-title"
>
  <img
    class="my-title-img"
    src="/images/ethereum.png"
  />
  <span
    class="my-title-label "
  >
    Etherbeam
  </span>
</span>
`;
